runtime/src/main/java/org/corfudb/runtime/collections/CorfuTable.java

```diff
@@ -233,8 +237,9 @@ public V get(@ConflictParameter Object key) {
     Collection<Entry<K, V>> getByIndex(@Nonnull Index.Name indexName, I indexKey) {
         String secondaryIndex = indexName.get();
         Map<Object, Map<K, V>> secondaryMap;
-        if (secondaryIndexes.containsKey(secondaryIndex) &&
-                ((secondaryMap = secondaryIndexes.get(secondaryIndex)) != null)) {
+        if ((secondaryIndexes.containsKey(secondaryIndex) &&
+                ((secondaryMap = secondaryIndexes.get(secondaryIndex)) != null)) || secondaryIndexesAliasToPath.containsKey(secondaryIndex)
+                && ((secondaryMap = secondaryIndexes.get(secondaryIndexesAliasToPath.get(secondaryIndex))) != null)) {
```

Comment:
dont we need a similar change in getByIndexAndFilter(), un/mapSecondaryIndexes()?

Reply:
good catch, yes those paths require this as well, will add..